src/Http/Request/Headers.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Request;
+
+use CultuurNet\UDB3\Http\ApiProblem\ApiProblem;
+use Negotiation\Accept;
+use Negotiation\Exception\Exception as NegotiationException;
+use Negotiation\Negotiator;
+use Psr\Http\Message\ServerRequestInterface;
+
+final class Headers
+{
+    private ServerRequestInterface $request;
+
+    public function __construct(ServerRequestInterface $request)
+    {
+        $this->request = $request;
+    }
+
+    /**
+     * Parses the Accept header of the request and determines the best Content-Type to use in the response based on a
+     * list of possible Content-Types.
+     *
+     * If there are one or more Accept lines and none matches the possible Content-Types, a 406 ApiProblem will be
+     * thrown.
+     *
+     * @param string[] $possibleContentTypes
+     *
+     * @throws ApiProblem
+     */
+    public function determineResponseContentType(array $possibleContentTypes): string
+    {
+        $acceptHeader = $this->request->getHeaderLine('accept');
+        if ($acceptHeader === '' || $acceptHeader === '*') {
+            return $possibleContentTypes[0];
+        }
+
+        $negotiator = new Negotiator();
+
+        $notAcceptable = ApiProblem::notAcceptable('Acceptable media types are: ' . implode(', ', $possibleContentTypes));
+
+        try {
+            /** @var Accept|null $mediaType */
+            $mediaType = $negotiator->getBest($acceptHeader, $possibleContentTypes);
```

Comment:
Lately I am sometimes hesitant to add new dependencies, would this be something we could implement?

Reply:
I started off by implementing it myself but it has to follow the [rfc7231 spec for content negotiation](https://datatracker.ietf.org/doc/html/rfc7231#section-5.3), which I'm no expert in. I agree about not adding too many dependencies, but this one seems pretty standardized (because of the spec it's based on) and it's abstracted away via the custom `Headers` class, so if we need to replace it sometime in the future it should be easy to just change the usages in that one custom class? (It's not like it would be spread out all over the codebase)

Reply:
To give a complex example from the spec that's not trivial to implement ourselves:

```
Accept: text/plain; q=0.5, text/html,
             text/x-dvi; q=0.8, text/x-c
```

We don't need these elaborate features like the `q` right now, but an API client should be able to expect that we can at least parse it without crashing. Which I could not guarantee with a custom implementation right now, because there's so many things to take into account with this header.

Reply:
Oh and if it helps, I picked this package because it has no dependencies on its own. So it's literally "just" one extra package. 😄

Reply:
Ok, no waterfall of extra packages 😀